06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js

```diff
@@ -0,0 +1,276 @@
+   * contain each book with their title and text.
+   * @returns {Array} - an Array that contains all the words in
+   * the book without duplication
+   */
+  static getAllWords(file) {
+    let words = '';
+    for (let i = 0; i < file.length; i += 1) {
+      words += `${file[i].text} `;
+    }
+    words = this.tokenize(words).slice(0, words.length - 1);
+    words = this.removeDuplicateWords(words);
+
+    return words.slice(0, words.length);
+  }
+
+  /**
+   * This function returns an array that consists of all the words in each book
+   * [[], [], ]
+   * @static
+   * @param {Array} file - contains an array of objects that
+   * contain each book with their title and text.
+   * @returns {Array} - an array that consist of all the words in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static getAllWordsInEachBook(file) {
+    const doc = [];
+    let bookContent;
+    for (let i = 0; i < file.length; i += 1) {
+      bookContent = `${file[i].text} `;
+      bookContent = this.tokenize(bookContent);
+      bookContent = this.removeDuplicateWords(bookContent);
+      doc.push(bookContent.slice(0, bookContent.length));
+    }
+    return doc;
+  }
+
+  /**
+   * This function takes two arrays and checks
+   * if each of the words in the first array is present in the second array
+   * @static
+   * @param {Array} words - contains all the words
+   * in the book without duplication
+   * @param {Array} wordsInEachBook - contains all the  words
+   * in a given document
+   * @returns {Array} - An array that consist of
+   * booleans to indicate if the words are present in each book
+   * e.g [true, false, false, false, false, true . . .]
+   * @memberOf InvertedIndexUtility
+   */
+  static checkOccurenceOfWords(words, wordsInEachBook) {
+    const checked = [];
+
+    words.forEach((word) => {
+      for (let i = 0; i < wordsInEachBook.length; i += 1) {
+        if (wordsInEachBook[i].indexOf(word) === -1) {
+          checked.push(false);
+        } else {
+          checked.push(true);
+        }
+      }
+    });
+    return checked;
+  }
+
+  /**
+   * This methods returns an array that consists of
+   * sub arrays of the word and booleans that indicate
+   *  if the word is present in the book
+   * e.g [[true, false, false], [false, false, true], [true, true, true]]
+   * @static
+   * @param {Array} checked - This array consist of
+   * booleans to indicate if the words are present in each book
+   * e.g [true, false, false, false, false, true . . .]
+   * @param {Integer} bookCount - The number of the book present in the file
+   * @returns {Array} - An Array that consist of sub arrays of each word
+   * and a boolean to indicate their presence in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static getWordOccurenceForEachBook(checked, bookCount) {
+    const result = [];
+    const stopIndex = Math.ceil(checked.length / bookCount);
+
+    for (let i = 0; i <= stopIndex; i += 1) {
```

Comment:
This right here is a violation of the Airbnb Style Guide. See [Link](https://github.com/airbnb/javascript#iterators--nope)